internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -729,6 +736,19 @@ func (r *ClusterDeploymentsReconciler) updateIfNeeded(ctx context.Context,
 		update = true
 	}
 
+	if clusterInstall.Spec.DiskEncryption != nil {
+		params.DiskEncryption = &models.DiskEncryption{}
+		if cluster.DiskEncryption == nil { // true when current cluster configuration does not include disk encryption
```

Comment:
In REST API, we always have a DiskEncryption object (we set the default values in https://github.com/openshift/assisted-service/blob/a3923b08fe52a0fad469db5d3d3cc6eaa7cc33e2/internal/bminventory/inventory.go#L613), therefore, in order to check if the feature is disable we need to check that `enable_on == none`.

Can we align to the same default and same checks?

Reply:
Yes. will change that 👍
I see this is a new change from yesterday (https://github.com/openshift/assisted-service/pull/2887), coded this before it had a default value.

Reply:
It is older than that and related to https://github.com/openshift/assisted-service/pull/2743.
The PR you mentioned just fixed a bug related to it.

Reply:
I looked at this again, and I see that the cluster model refers to DiskEncryption as a pointer:

https://github.com/openshift/assisted-service/blob/c7a98ee79bf967ef9bd3b5a202da9450e52de86d/models/cluster.go#L74-L75

which tells me, it is possible for it to be passed as `nil`, so I argue that this if statement should start to protect the controller from nil pointer exception. Given a cluster that has **anything** as DiskEncryption config, it will not enter this if statement anyway. WDYT?

Reply:
I see that DiskEncryption in cluster won't be initialized in case not specified in NewClusterParams:
https://github.com/openshift/assisted-service/blob/1ecc16b2160b843cde1650161954eb4d95f40b6f/internal/bminventory/inventory.go#L639
So seems like indeed safer to keep this check and initialization logic.

Reply:
ack. Marking this as resolved.